crates/ansi-to-html/src/test_utils.rs

```diff
@@ -0,0 +1,327 @@
+struct StylizedText {
+    styles: Styles,
+    text: String,
+}
+
+impl StylizedText {
+    fn new(styles: Styles, text: String) -> Self {
+        Self { styles, text }
+    }
+
+    fn empty(styles: Styles) -> Self {
+        let text = String::new();
+        Self { styles, text }
+    }
+}
+
+#[derive(Debug, Default, PartialEq, Eq)]
+struct Styles {
+    bold: bool,
+    italic: bool,
+    underlined: bool,
+    crossed_out: bool,
+    spans: BTreeSet<Vec<Attr>>,
+}
+
+impl Styles {
+    fn new(raw_styles: &[RawStyle]) -> Self {
+        raw_styles
+            .iter()
+            .cloned()
+            .fold(Styles::default(), |styles, s| styles.apply(s))
+    }
+
+    #[must_use]
+    fn apply(mut self, raw_style: RawStyle) -> Self {
+        match raw_style {
+            RawStyle::Bold => self.bold = true,
+            RawStyle::Italic => self.italic = true,
+            RawStyle::Underlined => self.underlined = true,
+            RawStyle::CrossedOut => self.crossed_out = true,
+            RawStyle::Span(span) => _ = self.spans.insert(span),
+        }
+        self
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+struct Attr {
+    name: String,
+    value: String,
+}
+
+impl Attr {
+    fn new(attr: &Attribute) -> Self {
+        let Attribute {
+            name: QualName { prefix, ns, local },
+            value,
+        } = attr;
+        assert!(prefix.is_none());
+        assert_eq!(ns, "");
+
+        let name = local.to_string();
+        let value = value.to_string();
+        Self { name, value }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use proptest::prelude::*;
+
+    #[test]
+    fn sanity() {
+        let ansi_text = "\x1b[1mBold\x1b[31mRed and Bold";
+        let htmlified = convert(ansi_text).unwrap();
+        insta::assert_debug_snapshot!(interpret_html(&htmlified), @r#"
+        [
+            StylizedText {
+                styles: Styles {
+                    bold: true,
+                    italic: false,
+                    underlined: false,
+                    crossed_out: false,
+                    spans: {},
+                },
+                text: "Bold",
+            },
+            StylizedText {
+                styles: Styles {
+                    bold: true,
+                    italic: false,
+                    underlined: false,
+                    crossed_out: false,
+                    spans: {
+                        [
+                            Attr {
+                                name: "style",
+                                value: "color:var(--red,#a00)",
+                            },
+                        ],
+                    },
+                },
+                text: "Red and Bold",
+            },
+        ]
+        "#);
+    }
+
+    proptest! {
+        #[test]
+        fn proptest_opt_equiv(ansi_text in r"(.*\x1b\[([0-9];?)+m){3,}.*") {
+            assert_opt_equiv_to_no_opt(&ansi_text);
+        }
+    }
```

Comment:
We could drop this proptest and instead rely on running the fuzzer for a short amount of time in CI instead 🤔

I think I like that more